asset-transfer-private-data/chaincode-go/private_asset_transfer.go

```diff
@@ -0,0 +1,497 @@
+	}
+
+	// Save asset to private data collection
+	err = ctx.GetStub().PutPrivateData(assetCollection, assetInput.ID, assetJSONasBytes)
+	if err != nil {
+		return fmt.Errorf("failed to put Asset: %s", err.Error())
+	}
+
+	// Save asset details to collection visible to owning organization
+	assetPrivateDetails := &AssetPrivateDetails{
+		ID:       			 	assetInput.ID,
+		AppraisedValue:   assetInput.AppraisedValue,
+	}
+
+	assetPrivateDetailsAsBytes, err := json.Marshal(assetPrivateDetails) // marshal asset details to JSON
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	// get collection name for this organization
+	orgCollection, err := getCollectionName(ctx)
+
+	// put asset appraised value into owners org specific private data collection
+	err = ctx.GetStub().PutPrivateData(orgCollection, assetInput.ID, assetPrivateDetailsAsBytes)
+	if err != nil {
+		return fmt.Errorf("failed to put asset private details: %s", err.Error())
+	}
+	return nil
+}
+
+// AgreeToTransfer is used by the potential buyer of the asset to agree to the
+// asset value. The agreed to appraisal value is stored in the buying orgs
+// org specifc collection, while the the buyer client ID is stored in the asset collection
+// using a composite key
+func (s *SmartContract) AgreeToTransfer(ctx contractapi.TransactionContextInterface) error {
+
+	// Get ID of submitting client identity
+	clientID, err := ctx.GetClientIdentity().GetID()
+	if err != nil {
+		return fmt.Errorf("failed to get verified OrgID: %s", err.Error())
+	}
+
+	// value is private, therefore it gets passed in transient field
+	transMap, err := ctx.GetStub().GetTransient()
+	if err != nil {
+		return fmt.Errorf("Error getting transient: " + err.Error())
+	}
+
+	// persist the JSON bytes as-is so that there is no risk of nondeterministic marshaling.
+	valueJSONasBytles, ok := transMap["asset_value"]
+	if !ok {
+		return fmt.Errorf("asset_value key not found in the transient map")
+	}
+
+	// unmarshal the tranisent map to get the asset ID.
+	var valueJSON AssetPrivateDetails
+	err = json.Unmarshal(valueJSONasBytles, &valueJSON)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+	}
+
+	// do some error checking since we get the chance
+	if len(valueJSON.ID) == 0 {
+		return fmt.Errorf("asset_id field must be a non-empty string")
+	}
+	if valueJSON.AppraisedValue <= 0 {
+		return fmt.Errorf("AppraisedValue field must be a positive integer")
+	}
+
+	// get collection name for this organization
+	orgCollection, err := getCollectionName(ctx)
+
+	// put agreed value in the org specifc private data collection
+	err = ctx.GetStub().PutPrivateData(orgCollection, valueJSON.ID, valueJSONasBytles)
+	if err != nil {
+		return fmt.Errorf("failed to put asset bid: %s", err.Error())
+	}
+
+	// create agreeement where you indicate the identity has agreed to purchase
+	transferAgreeKey, err := ctx.GetStub().CreateCompositeKey("transferAgreement",[]string{valueJSON.ID})
```

Comment:
since this object_type is referred in both go files, better to refer to a const in both files

`const transferAgreementObjectType := "transferAgreement"`

Reply:
done. added: `const transferAgreementObjectType = "transferAgreement"`